### PR TITLE
DRY up models and resources

### LINKED ISSUE #63: DRY up Volunteer resource
## Issue
- [ ] Style (non-breaking style code)
- [ ] Testing (add testing to code)
- [ ] Bug fix (change which fixes an issue)
- [ ] Refactor (improvement to the code)
- [ ] New feature (change which adds functionality)
- [x] Chore
## Issue Summary

DRY up the code

### PR DESCRIPTION
**Pull request by adumortier**
## Who worked on this

- [x] Alexis
- [ ] Maria
- [ ] Ed
- [ ] Cody

## Type of change

- [ ] Style (Non-breaking Style code)
- [ ] Testing (Non-breaking Testing code)
- [x] Refactor
- [ ] New feature (change which adds functionality)

## Summary of the change.

Link to issues: 
https://github.com/track-basket/trackbasket_BE/issues/63
https://github.com/track-basket/trackbasket_BE/issues/64

This closes #63 #64 

## Checklist:

- [x] My code follows the Turing style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented my code if necessary

### CODE REVIEW

File: trackbasket_be/models/at_risk_user.py [hunk 1]
```diff
@@ -48,4 +48,9 @@ def delete_shoppinglists(self):
   def delete(self):
     db.session.delete(self)
     db.session.commit()
-    
\ No newline at end of file
+  
+  def return_store(self):
+    return self.store[0]
```
Comment (adumortier) [inline trackbasket_be/models/at_risk_user.py:53]: moved this to a helper method to improve readability

File: trackbasket_be/models/krogerservice.py [hunk 2]
```diff
@@ -1,115 +1,117 @@
...
+    if term == '':
+      return 'error'
+    json_items = cls.__item_api_request(term, nearest_store_id)
+    if json_items == []:
+      return 'error'
+    items = cls.__extract_items(json_items)
+    return { 'data': { 'id': 'items', 'attributes':  items } }
+
+  @classmethod
+  def __return_token(cls):
```
Comment (adumortier) [inline trackbasket_be/models/krogerservice.py:31]: made `return_token` and other methods private ('__'). Setting the `access_token` as a `global` variable we can access it from other methods

File: trackbasket_be/models/krogerservice.py [hunk 3]
```diff
@@ -1,115 +1,117 @@
...
 
   @classmethod
-  def closest_store(cls, zipcode):
-    token = Krogerservice.refresh_token()
-    number_results = 5
+  def __find_image(cls, item):
+    if 'images' in item.keys():
+      for image in item['images']:
+        if (('perspective' in image.keys()) and image['perspective'] == 'front'):
+          return next((size['url'] for size in image['sizes'] if size['size'] == 'medium'), None)
```
Comment (adumortier) [inline trackbasket_be/models/krogerservice.py:51]: finds and returns the first item in the iterable that matches the given condition

File: trackbasket_be/resources/shopping_list.py [hunk 4]
```diff
@@ -11,81 +11,46 @@ class ShoppingLists(Resource):
...
-          return {'data': {  "id": "shoppinglist","attributes":
-              { "name": store.name, "address": store.address, "city": store.city, "state": store.state, "zipcode": store.zipcode,
-                "storeId": store.location_id, "latitude_longitude": [float(store.latitude), float(store.longitude)], "status": shopping_list.status,
-                "created_date": shopping_list.created_date.strftime("%d/%m/%Y %H:%M:%S"),
-                "items": [item.json() for item in shopping_list.items] } } }, 200
-
+    if at_risk_user.shopping_lists == []:
+      return {'data': { 'id': 'shoppinglist', 'attributes': {'error': "No shopping list associated with this user"} } }, 400
+    shopping_list = at_risk_user.return_shopping_list()
+    store = at_risk_user.return_store()
```
Comment (adumortier) [inline trackbasket_be/resources/shopping_list.py:31]: used helper methods here to improve readability

File: trackbasket_be/resources/shopping_list.py [hunk 5]
```diff
@@ -11,81 +11,46 @@ class ShoppingLists(Resource):
...
-      return {'data': {  "id": "shoppinglist","attributes":
-            { "name": store.name, "address": store.address, "city": store.city, "state": store.state, "zipcode": store.zipcode,
-              "storeId": store.location_id, "latitude_longitude": [float(store.latitude), float(store.longitude)], "status": shopping_list.status,
-              "created_date": shopping_list.created_date.strftime("%d/%m/%Y %H:%M:%S"),
-              "items": [item.json() for item in shopping_list.items] } } }, 200
+    store = at_risk_user.return_store()
+    if request_data["items"] == []:
+      return {'data': { 'id': 'shoppinglist', 'attributes': {'error': "Shopping List can not be empty"} } }, 400
+    shopping_list = ShoppingList.create_shopping_list(at_risk_user.id, request_data['status'], datetime.now())
+    shopping_list.add_items(request_data["items"])
```
Comment (adumortier) [inline trackbasket_be/resources/shopping_list.py:21]: used helper methods here to improve readability

File: trackbasket_be/models/krogerservice.py [hunk 6]
```diff
@@ -1,115 +1,117 @@
 from flask_restful import Resource, reqparse, request
 from .at_risk_user import AtRiskUser
-import requests
-import os
-import json
+import requests, os, json
```
Comment (mronauli) [inline trackbasket_be/models/krogerservice.py:3]: nice  readable one-liner

File: trackbasket_be/models/krogerservice.py (hunk 2 above)
Comment (mronauli) [reply trackbasket_be/models/krogerservice.py:31]: interesting that we can still access private methods outside this class using 'name mangling'

File: trackbasket_be/models/shopping_list.py [hunk 7]
```diff
@@ -2,10 +2,49 @@
...
+    db.session.commit()
+
+  def update_status(self, status):
+    self.status = status
+    db.session.commit()
+
+  def update(self, data):
+    self.update_status(data['status'])
+    self.delete_existing_items()
+    self.add_items(data["items"])
```
Comment (mronauli) [inline trackbasket_be/models/shopping_list.py:43]: nice move to clean up the resource and put methods here

**Pull request merged**